test/e2e/cluster/create.go

```diff
@@ -110,6 +103,27 @@ func (c *Create) Run(ctx context.Context, test TestResources) error {
 		return fmt.Errorf("loading kubeconfig: %w", err)
 	}
 
+	k8sClient, err := kubernetes.NewForConfig(clientConfig)
+	if err != nil {
+		return fmt.Errorf("creating kubernetes client: %w", err)
+	}
+
+	podIdentityAddon := addon.PodIdentityAddon{
+		Addon: addon.Addon{
+			Cluster:       hybridCluster.Name,
+			Name:          podIdentityAddonName,
+			Configuration: "{\"daemonsets\":{\"hybrid\":{\"create\": true}}}",
+		},
+		Kubernetes:         k8sClient,
+		IAMClient:          c.iam,
+		PodIdentityRoleArn: stackOut.podIdentityRoleArn,
+	}
+
+	err = podIdentityAddon.Create(ctx, c.eks, c.logger)
+	if err != nil && !errors.IsType(err, &types.ResourceInUseException{}) {
```

Comment:
aren't you handling that error now in the `Create` method itself? it does seem like it's a good thing to hide if the intent is to make the create operation idempotent